python/mxnet/amp/loss_scaler.py

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+"""Dynamic loss scaler for AMP."""
+from ..ndarray import multi_all_finite
+from ..ndarray import ndarray as nd
+from .. import autograd as ag
+
+class LossScaler(object):
+    """Dynamic loss scaler for AMP.
+
+    Properties
+    ----------
+    loss_scale : float
+        The current loss scale
+    """
+    def __init__(self):
+        self._loss_scale = 2.**16
+        self._next_loss_scale = self._loss_scale
+        self._max_loss_scale = 2.**24
+        self._scale_seq_len = 2000
```

Comment:
Is there any time an user wants to tweak seq_len for specific models? I am just curious because the fairseq impl has those options

Reply:
I don't think there is any guidance on what it should be besides "it should not happen too often", so that if it goes up and then immediately hits overflow, not many of the batches are lost.